08_analysis_tools/CalibrateVaneOffsets.m

```diff
@@ -0,0 +1,119 @@
+% / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / /
+% Calibrate Airflow Vane Mounting Offsets
+% / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / /
+
+% !!-- This script must be run *after running the logconv.m script. --!!
+
+% NOTE: airspeed scale, tube pressure drop, and vane position mapping
+%       calibrations are assumed completed before running this script.
+
+% Nonlinear least squares solution:
+% min_x ||f(x,u)||
+% subject to
+% w_d = 0, mean slip = 0
+% where 
+% y = f(x,u)
+% y_ref = [vI_n, vI_e, vI_d, slip]
+% x = [b_aoa, b_slip, w_n, w_e]
+% u = [u, aoa, slip, phi, theta, psi]
+
+%% / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / / /
+% Calibration data should satisfy the following criteria:
+% - in air at nominal airspeed
+% - constant altitude
+% - executing "figure-8" with minimal bank for at least 2 cycles (hint: use
+%   stabilized modes for collecting this data) -- note both left and right
+%   turns are needed to make the assumption that average slip in nominal
+%   conditions is = zero.
+% - wind speeds are assumed to be constant within the seleted data
+% - no vertical wind is present (assumed zero)
+% NOTE: slip will be assumed = zero in nominal stabilized conditions, this
+% may not be true in reality, but with wind, the mounting offset is
+% otherwise unobservable.
+
+% start and end times (modify these)
+t_st_cal = 2115;
+t_ed_cal = 2160;
+
+clc;
+if (topics.airspeed.logged && ...
+        topics.vehicle_gps_position.logged && ...
+        topics.vehicle_attitude.logged && ...
+        topics.sensor_hall.logged && ...
+        topics.sensor_hall_01.logged)
+    if ~(params.cal_hall_rev.logged && ...
+            params.cal_hall_p0.logged && ...
+            params.cal_hall_p1.logged && ...
+            params.cal_hall_p2.logged && ...
+            params.cal_hall_p3.logged && ...
+            params.cal_hall_01_rev.logged && ...
+            params.cal_hall_01_p0.logged && ...
+            params.cal_hall_01_p1.logged && ...
+            params.cal_hall_01_p2.logged && ...
+            params.cal_hall_01_p3.logged)
+        % if hall calibration params were not present in the logs, they
+        % must be defined here to execute the mounting offset calibration.
+        
+        % aoa
+        paramvector.cal_hall_rev = timeseries(-1, 0);
+        paramvector.cal_hall_p0 = timeseries(-14694678, 0);
```

Comment:
Are those magic numbers your initial guesses?

Reply:
these are actually the calibrations for the first two vanes i made. i leave them as example for any manual inputs others may put -- but also because these first two vanes are likely the only ones that will need back dated inputs (i have logs with these vanes that did not yet have the call hall params)

Reply:
Ok, then maybe add a comment explaining that.